### PR TITLE
Add support for reading input values from a file for IREE tools

### PR DESCRIPTION
**Pull request by hanhanW**
- Move parsing methods to vm_util.
- Enable the feature for benchmark_module and run_module as well.

Fixes https://github.com/google/iree/issues/2304

### CODE REVIEW

File: iree/tools/vm_util.h [hunk 1]
```diff
@@ -40,6 +40,13 @@ StatusOr<std::vector<RawSignatureParser::Description>> ParseInputSignature(
 StatusOr<std::vector<RawSignatureParser::Description>> ParseOutputSignature(
     const iree_vm_function_t& function);
 
+// Parses the content in |filename| into a variant list of VM scalars and
+// buffers. See below for the format of scalars and buffers. The inputs are
```
Comment (antiagainst) [inline iree/tools/vm_util.h:44]: s/below/ParseToVariantList/

File: iree/tools/benchmark_module_main.cc [hunk 2]
```diff
@@ -103,10 +107,20 @@ Status Run(::benchmark::State& state) {
...
 
-  ASSIGN_OR_RETURN(
-      iree_vm_variant_list_t * inputs,
-      ParseToVariantList(input_descs, iree_hal_device_allocator(device),
-                         absl::GetFlag(FLAGS_inputs)));
+  iree_vm_variant_list_t* inputs;
+  if (!absl::GetFlag(FLAGS_inputs_file).empty()) {
+    if (!absl::GetFlag(FLAGS_inputs).empty()) {
+      return InvalidArgumentErrorBuilder(IREE_LOC)
+             << "Expected only one of inputs and inputs_flag is set";
```
Comment (GMNGeoffrey) [inline iree/tools/benchmark_module_main.cc:114]: Nit: grammar

```suggestion
             << "Expected only one of inputs and inputs_flag to be set";
```

File: iree/tools/run_mlir_main.cc [hunk 3]
```diff
@@ -298,9 +281,9 @@ Status EvaluateFunction(iree_vm_context_t* context,
              << "Expected only one of input_values_file_flag and "
                 "input_values_flag is set";
```
Comment (GMNGeoffrey) [inline iree/tools/run_mlir_main.cc:282]:
```suggestion
                "input_values_flag to be set";
```

File: iree/tools/run_module_main.cc [hunk 4]
```diff
@@ -105,10 +109,20 @@ Status Run() {
...
 
-  ASSIGN_OR_RETURN(
-      iree_vm_variant_list_t * inputs,
-      ParseToVariantList(input_descs, iree_hal_device_allocator(device),
-                         absl::GetFlag(FLAGS_inputs)));
+  iree_vm_variant_list_t* inputs;
+  if (!absl::GetFlag(FLAGS_inputs_file).empty()) {
+    if (!absl::GetFlag(FLAGS_inputs).empty()) {
+      return InvalidArgumentErrorBuilder(IREE_LOC)
+             << "Expected only one of inputs and inputs_flag is set";
```
Comment (GMNGeoffrey) [inline iree/tools/run_module_main.cc:116]:
```suggestion
             << "Expected only one of inputs and inputs_flag to be set";
```

File: iree/tools/vm_util.h (hunk 1 above)
Comment (hanhanW) [reply iree/tools/vm_util.h]: Done

File: iree/tools/benchmark_module_main.cc (hunk 2 above)
Comment (hanhanW) [reply iree/tools/benchmark_module_main.cc]: Done

File: iree/tools/run_mlir_main.cc (hunk 3 above)
Comment (hanhanW) [reply iree/tools/run_mlir_main.cc]: Done

File: iree/tools/run_module_main.cc (hunk 4 above)
Comment (hanhanW) [reply iree/tools/run_module_main.cc]: Done

**Pull request merged**